### PR TITLE
Channel widget

### PR DESCRIPTION
**Pull request by fdrgsp**
merge after #164 

Add a `channel widget` based on `Generic PresetWidget` #128 and remove it from `main_window.py`.

### CODE REVIEW

File: micromanager_gui/_gui_objects/_channel_widget.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        self.channel_wdg.setParent(None)
+        self.channel_wdg.deleteLater()
+
+    def _update_widget(self, channel_group):
+        self.channel_wdg = self._create_channel_widget(channel_group)
+        self.layout().addWidget(self.channel_wdg)
+        if isinstance(
+            self.channel_wdg, PresetsWidget
+        ) and not self._mmc.getCurrentConfig(channel_group):
+            self.channel_wdg._combo.setStyleSheet("color: magenta;")
```
Comment (ianhi) [inline micromanager_gui/_gui_objects/_channel_widget.py:90]:
```suggestion
```

Just let the PresetWidget handle this. I think this is incorrectly causing it to be magenta on creation.

File: micromanager_gui/_gui_objects/_channel_widget.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    def _on_channel_group_changed(self, new_channel_group: str):
+        """When Channel group is changed, recreate combo."""
+        self._clear_previous_device_widget()
+        self._update_widget(new_channel_group)
+
+    def _clear_previous_device_widget(self):
+        self.channel_wdg.setParent(None)
+        self.channel_wdg.deleteLater()
+
+    def _update_widget(self, channel_group):
```
Comment (ianhi) [inline micromanager_gui/_gui_objects/_channel_widget.py:84]: Just combine these three methods?

**Review by ianhi — CHANGES_REQUESTED** [review]
When you start it from the launch-dev script it seems to incorrectly mark the channel with magenta. otherwise looking pretty good!

File: micromanager_gui/_gui_objects/_channel_widget.py (hunk 1 above)
Comment (fdrgsp) [reply micromanager_gui/_gui_objects/_channel_widget.py]: @ianhi you were right. To better handle that I made a small change in the PresetsWidget (#163).

**Review by tlambert03 — APPROVED** [review]
LGTM

**Pull request merged**